scipy/stats/_continuous_distns.py

```diff
@@ -3025,8 +3058,230 @@ def alphanot1func(alpha, beta, TH, aTH, bTH, cosTH, tanTH, W):
+            [2.040, 2.149, 2.461, 2.886, 3.356],
+            [2.098, 2.244, 2.676, 3.265, 3.912],
+            [2.189, 2.392, 3.004, 3.844, 4.775],
+            [2.337, 2.634, 3.542, 4.808, 6.247],
+            [2.588, 3.073, 4.534, 6.636, 9.144]]
+    
+        # Table VII - nu_zeta = psi_5(alpha, beta)
+        nu_zeta_table = [ 
+            [0, 0.000, 0.000, 0.000, 0.000],
+            [0, -0.017, -0.032, -0.049, -0.064],
+            [0, -0.030, -0.061, -0.092, -0.123],
+            [0, -0.043, -0.088, -0.132, -0.179],
+            [0, -0.056, -0.111, -0.170, -0.232],
+            [0, -0.066, -0.134, -0.206, -0.283],
+            [0, -0.075, -0.154, -0.241, -0.335],
+            [0, -0.084, -0.173, -0.276, -0.390],
+            [0, -0.090, -0.192, -0.310, -0.447],
+            [0, -0.095, -0.208, -0.346, -0.508],
+            [0, -0.098, -0.223, -0.380, -0.576],
+            [0, -0.099, -0.237, -0.424, -0.652],
+            [0, -0.096, -0.250, -0.469, -0.742],
+            [0, -0.089, -0.262, -0.520, -0.853],
+            [0, -0.078, -0.272, -0.581, -0.997],
+            [0, -0.061, -0.279, -0.659, -1.198]]
+    
+        psi_1 = interpolate.interp2d(nu_beta_range, nu_alpha_range, alpha_table, kind='linear')
+        psi_2 = interpolate.interp2d(nu_beta_range, nu_alpha_range, beta_table, kind='linear')
+        psi_2_1 = lambda nu_beta, nu_alpha: psi_2(nu_beta, nu_alpha) if nu_beta > 0 else -psi_2(-nu_beta, nu_alpha)
+        
+        phi_3 = interpolate.interp2d(beta_range, alpha_range, nu_c_table, kind='linear')
+        phi_3_1 = lambda beta, alpha: phi_3(beta, alpha) if beta > 0 else phi_3(-beta, alpha)
+        phi_5 = interpolate.interp2d(beta_range, alpha_range, nu_zeta_table, kind='linear')
+        phi_5_1 = lambda beta, alpha: phi_5(beta, alpha) if beta > 0 else -phi_5(-beta, alpha)
+        
+        # quantiles
+        p05 = np.percentile(data, 5)
+        p50 = np.percentile(data, 50)
+        p95 = np.percentile(data, 95)
+        p25 = np.percentile(data, 25)
+        p75 = np.percentile(data, 75)
+        
+        nu_alpha = (p95 - p05)/(p75 - p25)
+        nu_beta = (p95 + p05 - 2*p50)/(p95 - p05)
+    
+        if nu_alpha >= 2.439:
+            alpha = np.clip(psi_1(nu_beta, nu_alpha)[0], np.finfo(float).eps, 2.)
+            beta = np.clip(psi_2_1(nu_beta, nu_alpha)[0], -1., 1.)
+        else:
+            alpha = 2.0
+            beta = np.sign(nu_beta)
+        c = (p75 - p25) / phi_3_1(beta, alpha)[0] 
+        zeta = p50 + c*phi_5_1(beta, alpha)[0]
+        delta = np.clip(zeta-beta*c*np.tan(np.pi*alpha/2.) if alpha == 1. else zeta, np.finfo(float).eps, np.inf) 
+    
+        return (alpha, beta, delta, c)
+    
+    # Override fit method using quantile estimates.
+    def fit(self, data, *args, **kwds):
+        """Use McCullock 1986 method - Simple Consistent Estimators
+            of Stable Distribution Parameters
+        """
+        return self._fitstart(data, *args, **kwds)
```

Comment:
Is this correct? Isnt' the result from fitstart an approximation (it uses interpolation tables).

Reply:
It is really exciting to see stable laws implemented in the scipy. I have spend few years working with different density implementations. There are asymptotic expansions and intergral represenations, which performs pretty well and are more stable numerically-wise then fft. Also there are different parametrizations of the complex parameter in the exponent of characteristic function, which I believe are more intuitive. I guess it could be next development.

Reply:
@pv I removed the fit() override so this now uses MLE again. using interpolation tables as initial estimate. I simply reordered the test cases and it works; this could be a memory corruption somewhere. I suspect it's in the core library; if I can pin it down I'll raise a separate issue.